src/Helper/Settings.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Drupal\os2forms_organisation\Helper;
+
+use Drupal\Core\KeyValueStore\KeyValueFactoryInterface;
+use Drupal\Core\KeyValueStore\KeyValueStoreInterface;
+use Drupal\os2forms_organisation\Exception\InvalidSettingException;
+use Drupal\os2forms_organisation\Form\SettingsForm;
+
+/**
+ * General settings for os2forms_organisation.
+ */
+final class Settings {
+  /**
+   * The store.
+   *
+   * @var \Drupal\Core\KeyValueStore\KeyValueStoreInterface
+   */
+  private KeyValueStoreInterface $store;
+
+  /**
+   * The key value collection name.
+   *
+   * @var string
+   */
+  private $collection = 'os2forms_organisation.';
+
+  /**
+   * The setting keys.
+   *
+   * @var array|string[]
+   */
+  private array $keys = [
+    SettingsForm::TEST_MODE,
+    SettingsForm::AUTHORITY_CVR,
+    SettingsForm::CERTIFICATE,
+    SettingsForm::CACHE_EXPIRATION,
+    SettingsForm::ORGANISATION_SERVICE_ENDPOINT_REFERENCE,
+    SettingsForm::ORGANISATION_TEST_LEDER_ROLLE_UUID,
+    SettingsForm::ORGANISATION_PROD_LEDER_ROLLE_UUID,
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(KeyValueFactoryInterface $keyValueFactory) {
+    $this->store = $keyValueFactory->get($this->collection);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getAll(): array {
+    $values = $this->store->getMultiple(array_map([$this, 'buildKey'], $this->keys));
+
+    $vals = [];
+    foreach ($values as $key => $value) {
+      $vals[$this->unbuildKey($key)] = $value;
+    }
+
+    return $vals;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getKeys(): array {
+    return $this->keys;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function get(string $key, $default = NULL) {
+    return $this->store->get($this->buildKey($key), $default);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function set(string $key, $value): void {
+    $this->store->set($this->buildKey($key), $value);
+  }
+
+  /**
+   * Build key.
+   */
+  private function buildKey(string $key): string {
+    if (!in_array($key, $this->keys, TRUE)) {
+      throw new InvalidSettingException(sprintf('Invalid setting: %s', $key));
+    }
+    return $this->collection . $key;
+  }
+
+  /**
+   * Unbuild key.
+   */
+  private function unbuildKey(string $key): string {
+    return 0 === strpos($key, $this->collection) ? substr($key, strlen($this->collection)) : $key;
+  }
```

Comment:
These should be removed when using “keyvalue” (which is the correct way to do this). They are leftovers from use of “state”.

See https://github.com/itk-dev/os2forms_digital_post/blob/feature/DW-277-SF1601/src/Helper/Settings.php for an example.

Reply:
Fixed.